service/frontend/workflowHandler.go

```diff
@@ -3785,3 +3828,13 @@ func (wh *WorkflowHandler) validateSignalWithStartWorkflowTimeouts(
 
 	return nil
 }
+
+func (wh *WorkflowHandler) checkNamespaceMatch(requestNamespace string, tokenNamespace string, scope metrics.Scope) error {
+	if !wh.config.EnableTokenNamespaceEnforcement() {
+		return nil
+	}
+	if !strings.EqualFold(requestNamespace, tokenNamespace) {
```

Comment:
Namespaces are case sensitive. At least I was able to create two namespaces which are different in case only. It means  someone else might already have such namespaces created and this check will be backward compatible for them.

Reply:
namespace is case sensitive